src/Database/Expression/CommonTableExpression.php

```diff
@@ -0,0 +1,313 @@
+                $fields[$index] = $field = new IdentifierExpression($field);
+            }
+
+            if (!($field instanceof ExpressionInterface)) {
+                throw new InvalidArgumentException(sprintf(
+                    'The `$fields` argument must contain only instances of `%s`, or strings, `%s` given at index `%d`.',
+                    ExpressionInterface::class,
+                    getTypeName($field),
+                    $index
+                ));
+            }
+        }
+
+        $this->fields = $fields;
+
+        return $this;
+    }
+
+    /**
+     * Returns the modifiers to use for the CTE.
+     *
+     * @return \Cake\Database\ExpressionInterface[]|string[]
+     */
+    public function getModifiers(): array
+    {
+        return $this->modifiers;
+    }
+
+    /**
+     * Sets the modifiers to use for the CTE.
+     *
+     * @param \Cake\Database\ExpressionInterface[]|string[] $modifiers The modifiers to use for the CTE.
+     * @return $this
+     * @throws \InvalidArgumentException When one or more modifiers are of an invalid type.
+     */
+    public function setModifiers(array $modifiers)
+    {
+        foreach ($modifiers as $index => $modifier) {
+            if (
+                !($modifier instanceof ExpressionInterface) &&
+                !is_string($modifier)
+            ) {
+                throw new InvalidArgumentException(sprintf(
+                    'The `$modifiers` argument must contain only instances of `%s`, or strings, ' .
+                        '`%s` given at index `%d`.',
+                    ExpressionInterface::class,
+                    getTypeName($modifier),
+                    $index
+                ));
+            }
+        }
+
+        $this->modifiers = $modifiers;
+
+        return $this;
+    }
+
+    /**
+     * Returns the CTE query definition.
+     *
+     * @return \Cake\Database\ExpressionInterface|null
+     */
+    public function getQuery(): ?ExpressionInterface
+    {
+        return $this->query;
+    }
+
+    /**
+     * Sets the CTE query definition.
+     *
+     * @param \Cake\Database\ExpressionInterface $query The CTE query definition.
+     * @return $this
+     */
+    public function setQuery(ExpressionInterface $query)
+    {
+        $this->query = $query;
+
+        return $this;
+    }
+
+    /**
+     * Returns whether the CTE operates recursively.
+     *
+     * @return bool
+     */
+    public function isRecursive(): bool
+    {
+        return $this->recursive;
+    }
+
+    /**
+     * Sets whether the CTE operates recursively.
+     *
+     * @param bool $recursive Indicates whether the CTE query operates recursively.
+     * @return $this
+     */
+    public function setRecursive(bool $recursive)
+    {
+        $this->recursive = $recursive;
+
+        return $this;
+    }
+
+    /**
+     * Converts the Node into a SQL string fragment.
+     *
+     * @param \Cake\Database\ValueBinder $generator Placeholder generator object
```

Comment:
You can use `{@inheritDoc}` to just add throws tags.

Reply:
I've tried that but PHPCS still complained, trying it again it works. Might have been a different error and I just didn't pay attention.

Reply:
You will get a mix of errors depending on what else is in the docblock since cakephp-codesniffer can't tell what it *should* be.

Reply:
Now PHPCS is happy, but Stickler complains again. @markstory Does Stickler use the different rules than the default CS check?